src/ansi-c/CMakeLists.txt

```diff
@@ -98,6 +96,27 @@ add_library(ansi-c
     ${CMAKE_CURRENT_BINARY_DIR}/library-check.stamp
 )
 
+file(GLOB_RECURSE sources "*.cpp")
+file(GLOB_RECURSE headers "*.h")
+
+list(REMOVE_ITEM sources
+    "${CMAKE_CURRENT_SOURCE_DIR}/library/converter.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/file_converter.cpp"
+)
+
+add_library(ansi-c
+    ${sources}
+    ${headers}
+    ${BISON_parser_OUTPUTS}
+    ${FLEX_scanner_OUTPUTS}
+)
+
+set_source_files_properties(
```

Comment:
What is this doing?

Reply:
Yes, I'm a bit confused here too - this is removing converter and file_converter sources from the dependencies for ansi-c, which in someways makes sense. But should there be some sort of transitive dependency on the converter/file_converter tools so that if those tools get modified, anything that depends on them also gets rebuilt?

Reply:
I've updated this so that the custom commands depend on the executables they run.

I was confused about this, but it turns out CMake only adds a target-level dependency for commands which reference executables built in the project (i.e. the executable is built before the files it generates). CMake *doesn't* add a file-level dependency for files built by the executable. By adding an explicit `DEPENDS` to the custom command, we can force CMake to re-generate files that are created by the generator executables whenever the executables are changed.